dask/array/core.py

```diff
@@ -937,9 +937,15 @@ def store(sources, targets, lock=True, regions=None, compute=True,
 
         load_dsks_mrg = sharedict.merge(store_dsks_mrg, *load_dsks)
 
-        result = tuple(
-            Array(load_dsks_mrg, n, src.chunks, src.dtype) for n in load_names
-        )
+        result = []
+        for i in range(len(load_names)):
+            result.append(Array(
+                load_dsks_mrg,
+                load_names[i],
+                sources[i].chunks,
+                sources[i].dtype
+            ))
+        result = tuple(result)
```

Comment:
I prefer comprehensions when they're not too complicated (this one certainly isn't), as it's clear that the loop is just for building a result. They're also more efficient (not that this matters here).

```python
return tuple(Array(load_dsks_mrg, name, src.chunks, src.dtype)
             for name, src in zip(load_names, sources))
```

Reply:
Actually I'd prefer to keep this as a `for`-loop if it's all the same. I already messed up this code once and would much rather keep it in a form that is clear to me what it does. Hope that makes sense.

Reply:
Could you replace the indexing with `zip` at least?